File: sem3/matr/a.cpp
#include <bits/stdc++.h>

using namespace std;

int main() {
    freopen("schedule.in", "r", stdin);
    freopen("schedule.out", "w", stdout);

    long long n;
    cin >> n;
    vector<pair<long long, long long>> s;
    set<long long> z;
    for (long long i = 1; i <= n; ++i) {
        long long d, w;
        cin >> d >> w;
        s.push_back(make_pair(d, w));
        z.insert(i);
    }
    sort(s.begin(), s.end(), [](const pair<long long, long long> & lhs, const pair<long long, long long> & rhs) { return lhs.second > rhs.second || (lhs.second == rhs.second && lhs.first > lhs.first); });
    long long p = 0;
    for (long long i = 0; i < n; ++i) {
        auto j = z.upper_bound(s[i].first);
        if (j == z.end()) {
            continue;
        } else if (j == z.begin()) {
            p += s[i].second;
        } else {
            --j;
            z.erase(j);
        }
    }
    cout << p;
}

File: sem3/matr/a.py
import sys
sys.stdin = open('schedule.in', 'r')
sys.stdout = open('schedule.out', 'w')
n = int(input())
s = []
for _ in range(n):
    s.append(list(map(int, input().split())))
s.sort(key=lambda x: (-x[1], x[0]))
z = [-1] * n
p = 0
for i in range(n):
    for j in range(min(n, s[i][0]) - 1, -1, -1):
        if j == 0 and z[j] != -1:
            p += s[i][1]
            break
        elif z[j] == -1:
            z[j] = s[i][1]
            break
print(p)
